src/bilder/images/edxapp_v2/deploy.py

```diff
@@ -0,0 +1,420 @@
+    Consul,
+    ConsulAddresses,
+    ConsulConfig,
+    ConsulService,
+    ConsulServiceTCPCheck,
+)
+from bilder.components.hashicorp.consul.steps import proxy_consul_dns
+from bilder.components.hashicorp.consul_template.models import (
+    ConsulTemplate,
+    ConsulTemplateConfig,
+    ConsulTemplateTemplate,
+    ConsulTemplateVaultConfig,
+)
+from bilder.components.hashicorp.consul_template.steps import (
+    consul_template_permissions,
+)
+from bilder.components.hashicorp.steps import (
+    configure_hashicorp_product,
+    install_hashicorp_products,
+    register_services,
+)
+from bilder.components.hashicorp.vault.models import (
+    Vault,
+    VaultAgentCache,
+    VaultAgentConfig,
+    VaultAutoAuthAWS,
+    VaultAutoAuthConfig,
+    VaultAutoAuthMethod,
+    VaultAutoAuthSink,
+    VaultAutoAuthFileSink,
+    VaultConnectionConfig,
+    VaultListener,
+    VaultTCPListener,
+)
+from bilder.components.vector.models import VectorConfig
+from bilder.components.vector.steps import install_and_configure_vector
+from bilder.facts.has_systemd import HasSystemd
+from bilder.lib.linux_helpers import DOCKER_COMPOSE_DIRECTORY
+from bilder.lib.template_helpers import (
+    CONSUL_TEMPLATE_DIRECTORY,
+    place_consul_template_file,
+)
+from bridge.lib.magic_numbers import VAULT_HTTP_PORT
+from bridge.lib.versions import (
+    CONSUL_TEMPLATE_VERSION,
+    CONSUL_VERSION,
+    TUTOR_PERMISSIONS_VERSION,
+    VAULT_VERSION,
+)
+from bridge.secrets.sops import set_env_secrets
+
+from bilder.images.edxapp_v2.lib import OPENEDX_RELEASE, WEB_NODE_TYPE, node_type
+from bridge.settings.openedx.accessors import fetch_application_version
+from bridge.settings.openedx.types import OpenEdxApplication
+
+VERSIONS = {
+    "consul": os.environ.get("CONSUL_VERSION", CONSUL_VERSION),
+    "consul-template": os.environ.get(
+        "CONSUL_TEMPLATE_VERSION", CONSUL_TEMPLATE_VERSION
+    ),
+    "tutor_permissions": os.environ.get(
+        "TUTOR_PERMISSIONS_VERSION", TUTOR_PERMISSIONS_VERSION
+    ),
+    "vault": os.environ.get("VAULT_VERSION", VAULT_VERSION),
+}
+TEMPLATES_DIRECTORY = Path(__file__).resolve().parent.joinpath("templates")
+FILES_DIRECTORY = Path(__file__).resolve().parent.joinpath("files")
+
+set_env_secrets(Path("consul/consul.env"))
+
+vector_config = VectorConfig(is_docker=True, use_global_log_sink=True)
+vector_config.configuration_templates[
+    TEMPLATES_DIRECTORY.joinpath("vector", "edxapp_parsing.yaml.j2")
+] = {}
+
+# Setup some environment variables that will be pulled in by docker / docker-compose
+EDX_INSTALLATION_NAME = os.environ.get("EDX_INSTALLATION", "mitxonline")
+edx_version = fetch_application_version(
+    OPENEDX_RELEASE, EDX_INSTALLATION_NAME, OpenEdxApplication.edxapp
+)
+files.put(
+    name=f"Setting the EDXAPP_DOCKER_TAG env var to {OPENEDX_RELEASE}-{EDX_INSTALLATION_NAME}",  # noqa: E501
+    src=io.StringIO(f"{OPENEDX_RELEASE}-{EDX_INSTALLATION_NAME}"),
```

Comment:
Should we resolve this to the specific hash of the image at build time to ensure we get exactly what we think we're getting? Particularly if we cycle instances in an ASG in production, after building a new image in CI?

Reply:
We can probably just pass that hash as an env var during the build, propagated via the pipeline